ocp_resources/performance_profile.py

```diff
@@ -0,0 +1,121 @@
+            net (dict, optional): Defines a set of network related features. Expected key-values are:
+                devices: (dict, optional). Expected key-values are:
+                    deviceID (str, optional): Network device ID represented by 16 digit hexadecimal number.
+                    interfaceName (str, optional): Network device name to be matched.
+                    vendorID (str, optional): Network device vendor ID represented by 16 digit hexadecimal number.
+                userLevelNetworking (boolean, optional, default: False): when enabled sets either all or specified network devices queue size to the amount of reserved CPUs.
+            node_selector (str, required): Defines the node label to use in the NodeSelectors of resources like Tuned created by the operator.
+            numa (dict, optional): Defines options related to topology aware affinities. Expected key-values are:
+                topologyPolicy (str, optional): Name of the policy when TopologyManager is enabled.
+            real_time_kernel (dict, optional):  Defines a set of real time kernel related parameters. Expected key-values are:
+                enabled (boolean, optional, default: False): Defines if the real time kernel packages should be installed.
+            workload_hints (dict, optional): Defines hints for different types of workloads. Expected key-values are:
+                highPowerConsumption (boolean, optional, default: False): Defines if the node should be configured for high power consumption.
+                mixedCpus (boolean, optional, default: False): Enables mixed-cpu-node-plugin on the node if set.
+                perPodPowerManagement (boolean, optional, default: False): Defines if the node should be configured for per pod power management.
+                realTime (boolean, optional, default: True): Defines if the node should be configured for real time workload.
+        """
+
+        super().__init__(**kwargs)
+        self.additional_kernel_args = additional_kernel_args
+        self.cpu = cpu
+        self.globally_disable_irq_load_balancing = globally_disable_irq_load_balancing
+        self.hugepages = hugepages
+        self.machine_config_label = machine_config_label
+        self.machine_config_pool_selector = machine_config_pool_selector
+        self.net = net
+        self.node_selector = node_selector
+        self.numa = numa
+        self.real_time_kernel = real_time_kernel
+        self.workload_hints = workload_hints
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if not self.cpu:
+                raise ValueError("Passing yaml_file or parameter 'cpu' is required")
+            if not self.cpu.isolated:
+                raise ValueError("Passing parameter 'isolated' for 'cpu' parameter is required")
+            if not self.cpu.reserved:
+                raise ValueError("Passing parameter 'reserved' for 'cpu' parameter is required")
```

Comment:
should be removed; we only verify what is sent by the user (e.g `cpu`)

Reply:
done